### PR TITLE
Added restaurant model and changed the post and user models

### PR DESCRIPTION
**Pull request by dpowley**

### CODE REVIEW

**Comment by njbrake** [comment]
This is probably something worthwhile to discuss or receive some clarity from Sam/James/Bekah: this pull uses food specific mutations ("Restaurant"). Are we assuming that the experiences will be confined to restaurants for the time being? Or should they be named to something more experience agnostic ("Business" || "Company" || "ExperiencePlace")?

File: server/schema/mutations.js [hunk 1]
```diff
@@ -5,29 +5,42 @@ const {
...
+                lastName: { type: new GraphQLNonNull(GraphQLString) },
+                gender: { type: new GraphQLNonNull(GraphQLString) },
+                birthDate: { type: new GraphQLNonNull(GraphQLInt) },
+                phoneNumber: { type: GraphQLInt },
+                location: { type: new GraphQLNonNull(GraphQLString) },
+                lat: { type: new GraphQLNonNull(GraphQLFloat) },
+                long: { type: new GraphQLNonNull(GraphQLFloat) },
+                bio: { type: GraphQLString },
+                profilePicture: { type: GraphQLString },
+                experiencePreferences: { type: new GraphQLList(GraphQLString) }
```
Comment (coopersamuel) [inline server/schema/mutations.js]: I want to create some GraphQL input types to clean these up a bit, but this is good for now!

File: server/models/post.js [hunk 2]
```diff
@@ -11,19 +12,87 @@ const PostSchema = new Schema({
...
+    await Restaurant.findById({ "_id": doc.restaurant }, async function(err, obj) { 
+        await Restaurant.findByIdAndUpdate(doc.restaurant, { $set: { avgRating: ((obj.toObject().ratingCount * obj.toObject().avgRating) - doc.review)/(obj.toObject().ratingCount - 1) } }
+   )});
+
+    // ratingCount 
+    await Restaurant.findByIdAndUpdate(doc.restaurant, { $inc: { ratingCount: -1 } });
+    next();
+});
+
+
```
Comment (coopersamuel) [inline server/models/post.js]: I think that bug you and I were looking at is in this block of code. Let's pair on this soon

File: server/schema/mutations.js (hunk 1 above)
Comment (dpowley) [reply server/schema/mutations.js]: Can you give some more clarification on this?

**Pull request merged**